plugins/woocommerce/src/Blocks/BlockTypes/MiniCart.php

```diff
@@ -461,7 +460,29 @@ protected function render_experimental_iapi_mini_cart( $attributes, $content, $b
 		$product_count_visibility = isset( $attributes['productCountVisibility'] ) ? $attributes['productCountVisibility'] : 'greater_than_zero';
 		$wrapper_classes          = sprintf( 'wc-block-mini-cart wp-block-woocommerce-mini-cart %s', $classes_styles['classes'] );
 		$wrapper_styles           = $classes_styles['styles'];
-		$template_part_contents   = $this->get_template_part_contents();
+		$template_part_contents   = $this->get_template_part_contents( 'experimental-iapi-mini-cart' );
+
+		$cart            = $this->get_cart_instance();
+		$cart_item_count = $cart ? $cart->get_cart_contents_count() : 0;
+		$cart_item_text  = '';
+
+		if ( 'always' === $product_count_visibility ) {
+			$cart_item_text = $cart_item_count;
+		} elseif ( 'never' !== $product_count_visibility && $cart_item_count > 0 ) {
+			$cart_item_text = $cart_item_count;
+		}
```

Comment:
Why do we need to do this? Wouldn't be a better approach to not render the badge `span` at all if its visibility is `never`?

Reply:
We don't render if its never so this condition is not really needed. I believe now that I've refactored badgeIsVisible this should be solved.